

Welcome to the documentation for BOFdat
=======================================

BOFdat is a package to generate biomass objective function (BOF) for genome-scale models (GEM) from experimental data. The definition of the BOF impacts the quality of phenotypic predictions generated by the model. Often times the BOF is defined using another reconstruction work or another organism. This package is designed to help you stand out of the pack by using experimental data specific to your organism and increase the value of your model. The easiest way to use BOFdat is to download it via pip install (see install). A full usage example, reconstructing the BOF for *Escherichia coli* is available on GitHub, where the entire source code is also available for developers.

General Usage and implementation details
----------------------------------------

BOFdat is conceived as a workflow for the construction of the BOF divided in three independent steps. Following steps 1 to 3 in order should proivde the users with a BOF that matches experimental growth rates and yields a high accuracy of essentiality prediction. Following the workflow is not strictly enforced. We briefly detail each step of the workflow:

**Step1:** Generate the stoichiometric coefficients for major macromolecules.

**Step2:** Find inorganic ions and coenzymes and generate stoichiometric coefficients.

**Step3:** Find the remaining species-specific metabolic objective and generate stoichiometric coefficients. 

We found that the macromolecular weight fractions (MWF) are the major determinant of the stoichiometric coefficients. Hence, we advise modellers to query detailed composition of their organism. This signifies determining the absolute dry weight of the cell and the percentage dry weight occupied by the major components: proteins, RNA, lipids, DNA and glycans. For more information on how to obtain this composition experimentally we refer to: Beck AE, Hunt KA, Carlson RP. Measuring Cellular Biomass Composition for Computational Biology Applications. Processes. Multidisciplinary Digital Publishing Institute; 2018;6: 38. doi:10.3390/pr6050038

To generate accurate growth rate prediction, obtaining growth and non-growth associated maintenance (GAM and NGAM) ATP costs is usually the most important parameter. Growth rate data on different substrate with quantification of substrate consumption rate and metabolic waste production rate can be incorporated in Step1 of BOFdat to generate the experimentally determined maintenance costs. We provide an example of such data and, for further experimental reference, we suggest referring to: Monk JM, Lloyd CJ, Brunk E, Mih N, Sastry A, King Z, et al. iML1515, a knowledgebase that computes Escherichia coli traits. Nat Biotechnol. 2017;35: 904–908. doi:10.1038/nbt.3956

Coenzymes and inorganic ions can be hard to identify experimentally. BOFdat Step2 takes care of identifying these ions by comparing the your reconstructed model with the table of universal biomass components generated by Rocha and colleagues: Xavier JC, Patil KR, Rocha I. Integration of Biomass Formulations of Genome-Scale Metabolic Models with Experimental Data Reveals Universally Essential Cofactors in Prokaryotes. Metab Eng. 2017;39: 200–208. doi:10.1016/j.ymben.2016.12.002

The metabolite composition of the BOF reflects the growth requirements of the organism and therefore impacts the gene essentiality prediction of the model. BOFdat provides an unbiased way to determine these metabolic objectives that uses experimental essentiality data. This is useful for users that which to increase or validate the essentiality prediction formualted by the model. This requires a genome-wide essentiality dataset. Usually this type of data is obtained with high-throughput insertion of transposon or, more rarely, with genome-wide single gene knock-out. For more details on experimental high-throughput transposon insertion see: van Opijnen T, Bodi KL, Camilli A. Tn-seq: high-throughput parallel sequencing for fitness and genetic interaction studies in microorganisms. Nat Methods. 2009;6: 767–772.


A good example of genome-wide single gene knock-out is the "Keio collection" for *E. coli*, see: Baba T, Ara T, Hasegawa M, Takai Y, Okumura Y, Baba M, et al. Construction of Escherichia coli K-12 in-frame, single-gene knockout mutants: the Keio collection. Mol Syst Biol. 2006;2: 2006.0008. 

Installation
------------

BOFdat can be installed using pip::
   
   pip install BOFdat


Citation:
---------

If BOFdat helped in the construction of your biomass objective function or the identification of metabolic objectives, please include the following citation to your work:
Lachance J-C, Monk JM, Lloyd CJ, Seif Y, Palsson BO, Rodrigue S, et al. BOFdat: generating biomass objective function stoichiometric coefficients from experimental data [Internet]. bioRxiv. 2018. p. 243881. doi:10.1101/243881
https://www.biorxiv.org/content/10.1101/243881v1

Contents
--------

.. toctree::
   :numbered:
   :maxdepth: 2
   
   Getting_started
   BOFdat_step1
   BOFdat_step2
   BOFdat_step3
   API
   license
   help   
   

* :ref:`genindex`
* :ref:`modindex`
* :ref:`search`
